packages/flame/lib/src/effects2/rotate_effect.dart

```diff
@@ -0,0 +1,60 @@
+import 'effect_controller.dart';
+import 'transform2d_effect.dart';
+
+/// Rotate a component around its anchor.
+///
+/// Two constructors are provided:
+///   - [RotateEffect.by] will rotate the target by the specified `angle`
+///     relative to its orientation at the onset of the effect. For example,
+///     rotating by `angle = tau/4` will turn the component 90° clockwise
+///     relative to its initial direction;
+///   - [RotateEffect.to] will rotate the target to the fixed orientation
+///     specified by the `angle`. For example, rotating to `angle = tau/4` will
+///     turn the component to look East regardless of its initial bearing.
+///
+/// This effect applies incremental changes to the component's angle, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class RotateEffect extends Transform2DEffect {
+  RotateEffect.by(double angle, EffectController controller)
+      : _angle = angle,
+        super(controller);
+
+  factory RotateEffect.to(double angle, EffectController controller) {
+    return _RotateToEffect(angle, controller);
+  }
+
+  /// The magnitude of the effect: how much the target should turn as the
+  /// progress goes from 0 to 1.
+  double _angle;
+
+  /// This variable keeps track of the effect's `progress` from the last time
+  /// that the `apply()` method was called.
+  double _lastProgress = 0;
```

Comment:
Maybe this is something that `Transform2DEffect` should keep track of? Will it not be used by more effects later?

Reply:
I guess you're right, that would save us from having to redefine `reset()` method too.